packages/core/src/types.ts

```diff
@@ -499,11 +566,13 @@ export interface StateNodeConfig<
   /**
    * The mapping of state node keys to their state node configurations (recursive).
    */
-  states?: StatesConfig<TContext, TEvent, TAction> | undefined;
+  states?: StatesConfig<TContext, TEvent, TAction, TActor> | undefined;
   /**
    * The services to invoke upon entering this state node. These services will be stopped upon exiting this state node.
    */
-  invoke?: SingleOrArray<string | InvokeConfig<TContext, TEvent>>;
+  invoke?: SingleOrArray<
+    TActor['src'] | InvokeConfig<TContext, TEvent, TActor>
```

Comment:
Passing a string to `invoke` is still supported - but it seems that we only have 4~ tests using this "flavor" and none of them is specifically testing this flavor. Should we continue supporting this?

Reply:
Especially if it's simpler to type, I don't think we should continue supporting it. Quick reasons:

- Only ~one~ two ways to invoke: `invoke: {}` and `invoke: [{}, {}, ...]` = good for consistency (besides `SingleOrArray`)
- Having `src: ...` subtly encourages adding `onDone` and `onError`, which are good things to add
- Simplifies things for future `machine-extractor`

Reply:
👍 let's do this in a followup PR